services/placement/placement.go

```diff
@@ -112,9 +115,10 @@ func (p *placement) SetInstances(instances []services.PlacementInstance) service
 		sort.Sort(ByIDAscending(instances))
 	}
 
-	p.instancesByShard = instancesByShard
-	p.instances = instancesMap
-	return p
+	newp := *p
```

Comment:
There is not a lot of places for that kind of replacement, I usually only call the setters when returning a placement in the end of a capacity change. I was planning to gradually change all semantics to that way since it's used in most other packages, but don't feel strongly either way. I'm gonna revert this.